src/taco-utils/telemetry.ts

```diff
@@ -379,15 +385,45 @@ module TacoUtility {
                 fs.writeFileSync(TelemetryUtils.telemetrySettingsFile, JSON.stringify(TelemetryUtils.telemetrySettings));
             }
 
-            private static getMachineId(): string {
+            private static getUniqueId(regValue: string, regHive: winreg.Hive, fallback: () => string): Q.Promise<any> {
+                var uniqueId: string;
+                var deferred: Q.Deferred<string> = Q.defer<string>();
+                if (os.platform() === "win32") {
+                    TelemetryUtils.getRegistryValue(TelemetryUtils.REGISTRY_SQMCLIENT_NODE, regValue, regHive)
```

Comment:
Rather than using deferreds, you can just return the promise chain here, and in the else branch you can return `Q.resolve(fallback())`